java/src/main/java/ai/onnxruntime/OnnxRuntime.java

```diff
@@ -5,175 +5,135 @@
+      }
+      System.load(libraryFilePath);
+      logger.log(Level.FINE, "Loaded native library '" + library + "' from specified path");
+      return;
+    }
 
-            if (filename != null && prefix.length() >= 3) {
-                File temp = File.createTempFile(prefix, suffix);
-                if (debugLogging) {
-                    logger.info("Writing " + path + " out to " + temp.getAbsolutePath());
-                }
-                temp.deleteOnExit();
-                if (!temp.exists()) {
-                    throw new FileNotFoundException("File " + temp.getAbsolutePath() + " does not exist.");
-                } else {
-                    byte[] buffer = new byte[1024];
-                    try (InputStream is = OnnxRuntime.class.getResourceAsStream(path)) {
-                        if (is == null) {
-                            throw new FileNotFoundException("File " + path + " was not found inside JAR.");
-                        } else {
-                            int readBytes;
-                            try (FileOutputStream os = new FileOutputStream(temp)) {
-                                while ((readBytes = is.read(buffer)) != -1) {
-                                    os.write(buffer, 0, readBytes);
-                                }
-                            }
-                            return temp.getAbsolutePath();
-                        }
-                    }
-                }
-            } else {
-                throw new IllegalArgumentException("The filename has to be at least 3 characters long.");
-            }
+    // 3) try loading from resources or library path:
+    // generate a platform specific library name
+    // replace Mac's jnilib extension to dylib
+    String libraryFileName = System.mapLibraryName(library).replace("jnilib", "dylib");
+    String resourcePath = "/ai/onnxruntime/native/" + libraryFileName;
+    File tempFile = tempDirectory.resolve(libraryFileName).toFile();
+    try(InputStream is = OnnxRuntime.class.getResourceAsStream(resourcePath)){
+      if (is == null) {
+        // 3a) Not found in resources, load from library path
+        logger.log(Level.FINE, "Attempting to load native library '" + library + "' from library path");
+        System.loadLibrary(library);
+        logger.log(Level.FINE, "Loaded native library '" + library + "' from library path");
+      } else {
+        // 3b) Found in resources, load via temporary file
+        logger.log(
+            Level.FINE,
+            "Attempting to load native library '" + library + "' from resource path " + resourcePath + " copying to " + tempFile);
+        byte[] buffer = new byte[1024];
+        int readBytes;
+        try (FileOutputStream os = new FileOutputStream(tempFile)) {
+          while ((readBytes = is.read(buffer)) != -1) {
+            os.write(buffer, 0, readBytes);
+          }
         }
+        System.load(tempFile.getAbsolutePath());
```

Comment:
So, if onnxruntime.dll has any dependencies that are not in the java.library.path, then this call will fail, right?

Reply:
I do not know precisely since I do not work in windows environments, but I believe the load call will load the file and whatever dependency configuration is set in the library file. Linux SO files can have relative, absolute, and rpath dependency locations. DLL's should probably have similar capabilities.

Reply:
It should check the system path for dependent libraries as it's basically a wrapper around the platform's dynamic library loader.